src/Raven.Server/Documents/Handlers/AttachmentHandler.cs

```diff
@@ -0,0 +1,209 @@
+using Voron.Exceptions;
+
+namespace Raven.Server.Documents.Handlers
+{
+    public class AttachmentHandler : DatabaseRequestHandler
+    {
+        [RavenAction("/databases/*/attachments", "GET")]
+        public async Task Get()
+        {
+            var documentId = GetQueryStringValueAndAssertIfSingleAndNotEmpty("id");
+            var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
+
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            using (context.OpenReadTransaction())
+            {
+                var attachment = Database.DocumentsStorage.GetAttachment(context, documentId, name);
+                if (attachment == null)
+                {
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                    return;
+                }
+
+                var etag = GetLongFromHeaders("If-None-Match");
+                if (etag == attachment.Etag)
+                {
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotModified;
+                    return;
+                }
+
+                var fileName = Path.GetFileName(attachment.Name);
+                HttpContext.Response.Headers["Content-Disposition"] = $"attachment; filename=\"{fileName}\"";
+                HttpContext.Response.Headers["Content-Type"] = attachment.ContentType.ToString();
+                HttpContext.Response.Headers[Constants.Headers.Etag] = "\"" + attachment.Etag + "\"";
+
+                JsonOperationContext.ManagedPinnedBuffer buffer;
+                using (context.GetManagedBuffer(out buffer))
+                using (var stream = attachment.Stream)
+                {
+                    var responseStream = ResponseBodyStream();
+                    var count = await stream.ReadAsync(buffer.Buffer.Array, 0, buffer.Length, Database.DatabaseShutdown);
+                    while (count > 0)
+                    {
+                        await responseStream.WriteAsync(buffer.Buffer.Array, 0, count, Database.DatabaseShutdown);
+                        count = await stream.ReadAsync(buffer.Buffer.Array, 0, buffer.Length, Database.DatabaseShutdown);
+                    }
+                }
+            }
+        }
+
+        [RavenAction("/databases/*/attachments", "PUT")]
+        public async Task Put()
+        {
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            {
+                var id = GetQueryStringValueAndAssertIfSingleAndNotEmpty("id");
+                var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
+                var contentType = GetStringQueryString("contentType", false) ?? "";
+
+                AttachmentResult result;
+                var tempPath = Path.Combine(Database.DocumentsStorage.Environment.Options.DataPager.Options.TempPath, $"attachment.{Guid.NewGuid():N}.put");
```

Comment:
Use Database.Configuration.Storage.TempPath that is a PathSetting and contains Combine that takes care of all platform related path issues?